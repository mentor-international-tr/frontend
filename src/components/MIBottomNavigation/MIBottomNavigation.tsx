import {
  BottomNavigation,
  BottomNavigationAction,
  withStyles,
} from "@material-ui/core";
import PropTypes from "prop-types";
import React, { useState, SyntheticEvent } from "react";
import classNames from "classnames";

const styles = {
  root: {},
};

interface NavItem {
  label: string;
  icon: React.ReactElement;
  clickHandler: React.ReactEventHandler<any>;
}

export interface MIBottomNavigation {
  classes: any;
  className?: string;
  navItems: NavItem[];
}

const MIBottomNavigation: React.SFC<MIBottomNavigation> = ({
  classes,
  className,
  navItems,
}) => {
  const [value, setValue] = useState(0);
  console.log(className);
  const handleChange = (event: any, newValue: number) => {
    setValue(newValue);
  };
  return (
    <BottomNavigation
      value={value}
      onChange={handleChange}
      showLabels
      className={classNames(classes.root, className)}
    >
      {navItems.map((item: NavItem) => {
        return (
          <BottomNavigationAction
            key={item.label}
            label={item.label}
            icon={item.icon}
            onClick={item.clickHandler}
          />
        );
      })}
    </BottomNavigation>
  );
};
MIBottomNavigation.propTypes = {
  classes: PropTypes.object.isRequired,
};
export default withStyles(styles)(MIBottomNavigation);
