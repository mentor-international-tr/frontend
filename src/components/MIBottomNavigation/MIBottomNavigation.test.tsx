import React from "react";
import { render, cleanup } from "react-testing-library";
import MIBottomNavigation from "./MIBottomNavigation";
import "jest-dom/extend-expect";

afterEach(cleanup);

describe("testing <MIBottomNavigation /> component", () => {
  it("creates snapshot", () => {
    const { asFragment } = render(
      <MIBottomNavigation navItems={[
          {
              label: "Favorites",
              icon: <i className="" />,
              clickHandler: (() => {})
          },
          {
            label: "Templates",
            icon: <i className="" />,
            clickHandler: (() => {})
        }
      ]}/>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
