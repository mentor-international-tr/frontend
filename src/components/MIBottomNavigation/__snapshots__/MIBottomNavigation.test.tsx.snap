// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <MIBottomNavigation /> component creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBottomNavigation-root-2 MIBottomNavigation-root-1"
  >
    <button
      class="MuiButtonBase-root-8 MuiBottomNavigationAction-root-3 MuiBottomNavigationAction-selected-4"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiBottomNavigationAction-wrapper-6"
      >
        <i
          class=""
        />
        <span
          class="MuiBottomNavigationAction-label-7 MuiBottomNavigationAction-selected-4"
        >
          Favorites
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-11"
      />
    </button>
    <button
      class="MuiButtonBase-root-8 MuiBottomNavigationAction-root-3"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiBottomNavigationAction-wrapper-6"
      >
        <i
          class=""
        />
        <span
          class="MuiBottomNavigationAction-label-7"
        >
          Templates
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-11"
      />
    </button>
  </div>
</DocumentFragment>
`;
